packages/core/src/signer/ckb/signerCkbMultisigReadonly.ts

```diff
@@ -0,0 +1,320 @@
+    return await Promise.all(
+      MULTISIG_SCRIPTS.map(async (script) => {
+        return await Script.fromKnownScript(client, script, args);
+      }),
+    );
+  }
+}
+
+/**
+ * A class extending Signer that provides access to a CKB multisig script.
+ * This class does not support signing operations.
+ * @public
+ */
+export class SignerCkbMultisigReadonly extends Signer {
+  get type(): SignerType {
+    return SignerType.CKB;
+  }
+
+  get signType(): SignerSignType {
+    return SignerSignType.CkbMultisigSecp256k1;
+  }
+
+  public readonly multisigInfo: MultisigInfo;
+
+  /**
+   * Creates an instance of SignerCkbMultisig.
+   *
+   * @param client - The client instance used for communication.
+   * @param privateKey - The private key associated with the signer.
+   * @param multisigInfoLike - The multisig information assembled from pubkeys, threshold, mustMatch and since.
+   */
+  constructor(client: Client, multisigInfoLike: MultisigInfoLike) {
+    super(client);
+
+    this.multisigInfo = MultisigInfo.from(multisigInfoLike);
+  }
+
+  async connect(): Promise<void> {}
+
+  async isConnected(): Promise<boolean> {
+    return true;
+  }
+
+  async getInternalAddress(): Promise<string> {
+    return this.getRecommendedAddress();
+  }
+
+  async getAddressObjSecp256k1(): Promise<Address> {
+    const multisigScript = await this.multisigInfo.defaultMultisigScript(
+      this.client,
+    );
+    return Address.fromScript(multisigScript, this.client);
+  }
+
+  async getAddressObjs(): Promise<Address[]> {
+    const multisigScripts = await this.multisigInfo.multisigScripts(
+      this.client,
+    );
+    return multisigScripts.map((script) =>
+      Address.fromScript(script, this.client),
+    );
+  }
+
+  async getRelatedScripts(
+    txLike: TransactionLike,
+  ): Promise<{ script: Script; cellDeps: CellDepInfo[] }[]> {
+    const tx = Transaction.from(txLike);
+    const multisig = await this.getAddressObjSecp256k1();
+
+    const scripts: { script: Script; cellDeps: CellDepInfo[] }[] = [];
+    for (const input of tx.inputs) {
+      const {
+        cellOutput: { lock },
+      } = await input.getCell(this.client);
+
+      if (scripts.some(({ script }) => script.eq(lock))) {
+        continue;
+      }
+
+      if (lock.eq(multisig.script)) {
+        const scriptInfo = await this.client.findKnownScript(lock);
+        if (scriptInfo) {
+          scripts.push({
+            script: lock,
+            cellDeps: scriptInfo.cellDeps,
+          });
+        }
+      }
+    }
+
+    return scripts;
+  }
+
+  /**
+   * Prepare multisig witness, if the existence of multisig witness is detected, nothing happens
+   *
+   * @param txLike - The transaction to prepare.
+   * @param scriptLike - The script to prepare.
+   * @returns A promise that resolves to the prepared transaction
+   */
+  async prepareTxMultisigWitness(
+    txLike: TransactionLike,
+    scriptLike: ScriptLike,
+  ) {
+    const tx = Transaction.from(txLike);
```

Comment:
If `txLike` is not an instance of `Transaction`, this will cause a new instance to be created, and all modifications here won't be able to affect the original argument.